src/runtimes/node/bundlers/nft/transpile.ts

```diff
@@ -52,22 +55,34 @@ export const transpileTS = async ({ bundle = false, config, format, name, path }
   // The version of ECMAScript to use as the build target. This will determine
   // whether certain features are transpiled down or left untransformed.
   const nodeTarget = getBundlerTarget(config.nodeVersion)
+  const bundleOptions: BuildOptions = {
+    bundle: false,
+  }
 
-  const bundleOptions: BuildOptions = bundle ? { bundle: true, packages: 'external' } : { bundle: false }
+  if (bundle) {
+    bundleOptions.bundle = true
+    bundleOptions.packages = 'external'
+
+    if (format === MODULE_FORMAT.ESM) {
```

Comment:
so we emit this shim in any case even if the srcs are esm?

Reply:
Yeah, I don't think we have an easy way to determine whether we have a CJS source or not. I mean, we could do it, but not sure if it's worth the hassle?